.github/workflows/templates-go.yaml

```diff
@@ -0,0 +1,112 @@
+name: Templates
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/templates-go.yaml"
+      - "templates/**"
+
+env:
+  TINYGO_VERSION: "0.33.0"
```

Comment:
I wonder if we should always be testing against the latest stable version of TinyGo?

It feels like a reasonable thing to do, but I could be convinced otherwise